src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -413,6 +428,38 @@ public function __construct(array $value = [])
 ');
     }
 
+    private function buildSetExtraKey(ClassBuilder $class): void
+    {
+        if (!$class->shouldAllowExtraKeys()) {
+            return;
+        }
+
+        $propertyName = $this->getExtraKeysProperty($class);
+
+        $class->addProperty($propertyName);
+
+        $class->addMethod('set'.substr($class->getName(), 0, -6).'ExtraKey', '
+/**
+ * @return $this
+ */
+public function NAME(string $key, $value): self
+{
+    if (null === $value) {
+        unset($this->PROPERTY[$key]);
+    } else {
+        $this->PROPERTY[$key] = $value;
+    }
+
+    return $this;
+}
+', ['PROPERTY' => $propertyName]);
+    }
+
+    private function getExtraKeysProperty(ClassBuilder $class): string
+    {
+        return lcfirst(substr($class->getName(), 0, -6).'ExtraKeys');
```

Comment:
Couldn't we just name this property to `_extraKeys`?

Reply:
@Nyholm Done, had to slightly modify the `addProperty` method.